Core/base elements content only (#1299)
massgov/mayflower: packages/core/stories/elements/Link/Link.stories.mdx

/>

# Links

Links are used to embed actions or pathways to more information in a sentence or on a page.

### Inline Link

- **Usage:** Use when you are linking just a few words of text or when you are doing inline links.
- **Style:** Inline links take the style of the surrounding text, adding only the blue color and an underline.

> <Canvas className="borderless">

Reviewer: ```suggestion
<Canvas>
```

Reviewer: not used

  <Story
    name="link">
      <p>Some content and then an <a href="#">inline link</a>, followed by more content.</p>
  </Story>
  <Story story={stories.linkCSS} />
</Canvas>

---

### Decorative Link

- **Usage:** Used for standalone links that require more emphasis, usually inside `ma__section-links`.

> - **Style:** the decorative link is slightly larger in size and appended with an arrow icon.

Reviewer: Can we also add the hover state to "Style" for decorative link

Author: Sure

  <Story story={stories.linkCalloutCSS} />
</Canvas>

---

> ### Link styled as a button

Reviewer: ```suggestion
### Link Styled as a Button
```

just nitpicking to be consistent as other headings

> - **Usage:** If absolutely necessary this can be used for calls-to-action if there are technical reasons not to use `ma__button`.

Reviewer: So "Link as a button" is using the `ma__button` className (expand the to show the HTML snippet), the difference here is that it's actually an anchor link `<a>` not a `<button>` (again, technically different elements).  Although there are use cases of in mayflower implementations, my opinion is that we should try to differentiate the styling between buttons and links, as they function differently, and that will help set expectations of the perceived behaviors.

Author: That makes sense

> - **Usage:** If absolutely necessary this can be used for calls-to-action if there are technical reasons not to use `ma__button`.

Reviewer: ```suggestion
- **Usage:** If absolutely necessary this can be used for calls-to-action if there are good design reasons not to use Callout Link. In general, links should be styled differently than buttons, to help set expectations of their different behaviors.
```
